goodai/ltm/memory.py

```
@@ -354,3 +354,36 @@ def _timestamp_search(self, origin_timestamp: float, lookahead_hops: int, lookba
                 current_index += direction
 
         return return_chunks
+
+    # Replace a memory by timestamp
+    def replace_content(self, target_timestamp: float, content: str,
+                        timestamp: float = None, keywords: list[str] = None,
+                        metadata: dict[str, Any] = None):
+
+        # Get textkeys that are associated with the timestamp
+        old_text_keys = []
+        mem: DefaultTextMemory = self.semantic_memory
+        seen = False
+
+        for chunk in reversed(mem.get_all_chunks()):
+            if chunk.timestamp == target_timestamp:
+                old_text_keys.extend(chunk.associated_keys)
+                seen = True
+            elif seen:
+                break
+
+        # There should only be one key after doing the chunks
+        key = old_text_keys[0]
+        for k in old_text_keys:
+            if key != k:
+                print(f"LTM: TextKeys are incosistient. Replacement will not go ahead.")
+                return
```

This could either fail (if no such timestamp is found) and also be replaced by a much simpler `if len(set(old_text_keys)) != 1`. I think that the LTM library should raise an exception instead of printing stuff.

---

I was thinking that the game would crash if this happens, but I could catch it on the director side

---

Yeah, I think that's better. LTM should work as a standard library, and the director is responsible for handling the exception.